iothub_client/inc/iothub_client_options.h

```diff
@@ -43,6 +43,9 @@ extern "C"
     */
     static const char* OPTION_C2D_KEEP_ALIVE_FREQ_SECS = "c2d_keep_alive_freq_secs";
 
+    //diagnostic sampling percentage value, [0-100]
+    static const char* OPTION_DIAGNOSTIC_SAMPLING_PERCENTAGE = "diag_sampling_percentage";
```

Comment:
What is the plan for the diagnostics? Will this eventually be applicable to all kinds of messages (Telemetry, Twin, ...)?

If this is restricted to Telemetry messages, this option should be more specific.

Reply:
e.g., OPTION_TELEMETRY_DIAGNOSTIC_SAMPLING_PERCENTAGE = "telemetry_...

Reply:
yes, finally will be applicable to all kinds of messages.